### DISCUSSION TITLE
对 KubeSphere 社区的 KubeKey 课题[2023年]的几点疑问

### DISCUSSION THREAD

**Discussion by LinuxSuRen** [summer-ospp/help#222] [General]
偶然发现几点对 KubeSphere 社区在参与开源之夏2023的 KubeKey 课题的几点疑问，希望能得到相关人员的解疑。

1. 一个包含了117个文件的[PR](https://github.com/kubesphere/kubekey/pull/2007)为什么没有公开 review 就直接合并了呢？
2. 代码没有合并到主干分支中，有没有相关的公开讨论说明，什么时候会合并到主干中？
3. 代码合并后超过一个多月的时间也没有在任何一个版本发布中包含相关功能，有没有计划什么时候发布，有没有 pre-release 版本之类的？

@shijilin0116 @liangzai006

**Comment by faweizhao26** [comment]
没太理解你说的没有公开 review 是什么意思，PR 肯定是经过 maintainer review 之后合并的。
至于你说的二三点，我们之前有开例会同步，录屏在 B 站可以找到。建议你提出疑问之前可以先去社区了解下情况。

**Reply by LinuxSuRen** [reply]
117 个文件，一个都不需要修改？一个comment都没？这个是公开 review？

PR 中也没有相关的视频链接。以上三点，我只是提出疑问，并没有否定。如果在 PR 中有相关链接描述，任何人就都可以很方便地看到上下文了。

> 建议你提出疑问之前可以先去社区了解下情况。

如果按照你的这个说法，难道有人提出疑问之前需要自行在所有平台（视频、文档）搜索才能拿到结果吗？无链接，无文档，难道你的意思有潜在 contributor 的话，也是建议这么去参与的吗？

正因为我也参与了不少该项目（KubeKey，包括从青云离职后也提交多个 PR）的 contribution 才会对这个项目很关心，但从你的回复中听到了“不爽”的感觉。如果维护者是在感觉我在挑刺，那么大可不必再回复。

**Reply by calvinyv** [reply]
非常感谢 rick 对 KubeSphere 持续关注和支持，我来解释一下：
1. KubeSphere 这次参与开源之夏的同学也是我们团队的一位新同学，因为我们希望开源的意识在我们团队里应该是一直传播下去的，而不是只有少数人参与，我们鼓励我们团队所有的人都参与开源，所以可能这位新同学并没有完全 follow 开源的最佳实践。
2. 参与开源之夏的学生本身也有学业任务，导师也有日常的工作，所以他们可能并没有去固定时间的进行公开化讨论，而是在日常随时随地的通过点对点的方式进行沟通了，只在一些关键里程碑节点，上了社区例会进行讨论。
3. 关于 pr 的问题，因为这个模块是 kubekey 上一个相对独立的全新模块，所以可能并没有 follow 代码管理的最佳实践开发，但我们确实可以保证代码经过 review 的。另外我个人觉得，这些不合理、这些瑕疵，本身就是开源最大的魅力之一，一切的一切也许并不完美，但都在通过开源的方式慢慢去迭代，参与开源的人也在这种开放的、轻松的氛围中去认识开源，得到了成长。
4. 我们非常感谢开源之夏，开源之夏最大的魅力在于持续的过程，正是由于开源之夏这么多年的坚持，还有像 rick 这样的导师，才让开源持续传播，让所有人看到了开源的开放和自由，KubeSphere 社区也会持续的参与贡献，我们在这个过程中也在不断地学习和成长，这次的收获也会成为下次的开源之夏的给养。

记得上个月在上海 kubecon 上，我在各个展台随便转的时候，突然有个年轻帅气的小伙拍我肩膀，介绍自己是参与我们开源之夏的同学，我当时真的特别高兴，正是由于开源的纽带，让我们这些本来没有关联的来自五湖四海的人产生了联系，这段联系就好像一滴水，也许将来就会是一片巨浪，这就是开源最有魅力的地方，感谢所有关注和参与开源之夏的人，KubeSphere 会继续努力做到更好。

**Reply by LinuxSuRen** [reply]
感谢 @calvinyv 认真的回复。正如你说说的，而且我也非常相信开源并不是要求当下完美，能够持续改进、持续追求卓越的社区（项目）一定是值得大家信任和追求的。
我猜测除了 KubeSphere 以外可能也还有其他社区、导师没有或者没有习惯 work-in-public 的方式。如果能有更多的人在意、讨论这个事情，相信后续开源之夏也会越办越好。
我参与过，以及听到过一些项目中，一些 PR 可能会有几十个来回的 comment，针对某个双方认为很重要的点进行讨论，互相给出一些理由、依据。（我认为）好的开源项目，应该会十分注重过程，结果只是顺其自然发生的。
我大概率也还会持续关注 KubeSphere 的发展情况，遇到好的和不好的案例，也还会“直言”讲出来。相信 @calvinyv 也一样认同 work-in-public 的力量和魅力。

**Comment by LinuxSuRen** [comment]
See also https://github.com/kubesphere/kubekey/issues/2260 and https://github.com/kubesphere/kubekey/tree/console

**Comment by LinuxSuRen** [comment]
23年的议题，今年25年了，代码依然没有合入主干。
